utils/zerovec/src/map/vecs.rs

```diff
@@ -35,6 +37,19 @@ pub trait ZeroVecLike<'a, T: ?Sized> {
     fn is_empty(&self) -> bool {
         self.len() == 0
     }
+
+    /// Convert T to a needle for searching
+    fn t_as_needle(t: &T) -> &T;
+
+    /// Compare this type with a `Self::GetType`. This must produce the same result as
+    /// if `g` were converted to `Self`
+    fn t_cmp_get(t: &T, g: &Self::GetType) -> Ordering;
+
+    /// Obtain a version of T suitable for serialization
+    ///
+    /// This uses a callback because it's not possible to return owned-or-borrowed
+    /// types without GATs
+    fn t_with_ser<R>(g: &Self::GetType, f: impl FnOnce(&T) -> R) -> R;
```

Comment:
Thought/Suggestion: Should we make a new trait with this one function in it and then bind it to `Self::GetType` ?

Reply:
No, because this puts more requirements on the implementor of ZMKV.

The goal of this PR is to move everything onto ZMLike, which is never implemented by consumers.